File: 백준/Bronze/17608. 막대기/막대기.py
import sys

a = int(sys.stdin.readline())

stk = []

for _ in range(a):
    hight = int(sys.stdin.readline())
    stk.append(hight)

m = 0
rst = 0

while stk:
    current = stk.pop()
    if current > m:
        rst += 1
        m = current

print(rst)

File: 백준/Bronze/17608. 막대기/막대기.cc
#include<bits/stdc++.h>
using namespace std;

int n;

int main(){

	stack<int> stk;
	cin >> n;
	
	for(int i=0;i<n;i++){
		int k =0;
		cin >> k;
		stk.push(k);
	}
	
	int a = stk.top();
	int rst = 1;
	stk.pop();
	while(!stk.empty()){
		if(stk.top()>a){
			rst ++;
			a = stk.top();
		}
		stk.pop();
	}
	cout << rst;

	return 0;
}